src/Node/HTTP2/Server.purs

```diff
@@ -0,0 +1,289 @@
+-- | Low-level bindings to the *Node.js* HTTP/2 Server Core API.
+-- |
+-- | ## Server-side example
+-- |
+-- | Equivalent to
+-- | https://nodejs.org/docs/latest/api/http2.html#server-side-example
+-- |
+-- | ```
+-- | key <- Node.FS.Sync.readFile "localhost-privkey.pem"
+-- | cert <- Node.FS.Sync.readFile "localhost-cert.pem"
+-- |
+-- | server <- createSecureServer (toOptions {key, cert})
+-- | _ <- onErrorServerSecure server Console.errorShow
+-- |
+-- | _ <- onceStreamSecure server \stream headers flags -> do
+-- |   respond stream
+-- |     (toHeaders
+-- |       { "content-type": "text/html; charset=utf-8"
+-- |       , ":status": 200
+-- |       }
+-- |     )
+-- |     (toOptions {})
+-- |   void $ Node.Stream.writeString (toDuplex stream)
+-- |     Node.Encoding.UTF8
+-- |     "<h1>Hello World</h1>"
+-- |     (\_ -> pure unit)
+-- |   Node.Stream.end (toDuplex stream) (\_ -> pure unit)
+-- |
+-- | listenSecure server
+-- |   (toOptions { port: 8443 })
+-- |   (pure unit)
+-- | ```
+module Node.HTTP2.Server
+  ( Http2Server
+  , createServer
+  , listen
+  , onceSession
+  , onStream
+  , onceStream
+  , onErrorServer
+  , closeServer
+  , onceCloseServer
+  , Http2SecureServer
+  , createSecureServer
+  , listenSecure
+  , onceSessionSecure
+  , onStreamSecure
+  , onceStreamSecure
+  , onErrorServerSecure
+  , closeServerSecure
+  , onceCloseServerSecure
+  , ServerHttp2Session
+  , respond
+  , localSettings
+  , closeSession
+  , ServerHttp2Stream
+  , session
+  , onceErrorStream
+  , pushAllowed
+  , pushStream
+  , onceTrailers
+  , additionalHeaders
+  , onceWantTrailers
+  , sendTrailers
+  , onceEnd
+  , toDuplex
+  , closeStream
+  ) where
+
+import Prelude
+
+import Data.Nullable (Nullable)
+import Effect (Effect)
+import Effect.Exception (Error)
+import Node.HTTP2 (Flags, HeadersObject, OptionsObject, SettingsObject)
+import Node.HTTP2.Internal as Internal
+import Node.Stream (Duplex)
+import Unsafe.Coerce (unsafeCoerce)
+
+-- | An HTTP/2 server with one listening socket for unencrypted connections.
+-- |
+-- | See [__Class: Http2Server__](https://nodejs.org/docs/latest/api/http2.html#class-http2server)
+foreign import data Http2Server :: Type
```

Comment:
I'm noticing that you split the two server types into two types (`Http2Server` & `Http2ServerSecure`). This has the effect of requiring you to also provide two implementations of the same function in a few cases (e.g. `closeServer`) that is the same FFI behind the scenes. When I looked at this, I wondered if the following design would work instead to reduce the number of duplicate functions.
```purs
data ConnectionType
foreign import data Encrypted :: ConnectionType
foreign import data Unencrypted :: ConnectionType

foreign import data Http2Server' :: ConnectionType -> Type
-- I think this is already nominal, but just in case...
type role Http2Server' nominal

type Http2Server = Http2Server' Unencrypted
type Http2ServerSecure = Http2Server' Enencrypted

-- These need to be duplicated because they're returning
-- back a different value and their FFIs are different
foreign import createServer :: OptionsObject -> Effect Http2Server
foreign import createSecureServer :: OptionsObject -> Effect Http2SecureServer

-- However, this does not need to be duplicated:
closeServer
  :: forall connectionType
   . Http2Server' connectionType -> Effect Unit -> Effect Unit
```